maven-setup/src/main.ts

```diff
@@ -59,4 +91,32 @@ function configureEnvironment(): void {
     core.exportVariable(ENV_KEY_MAVEN_OPTS, mavenOptsNew)
 }
 
-run()
+async function downloadFile(url: string, downloadFolder: string): Promise<string> {
+    const fileName = path.basename(url);
+    const filePath = path.join(downloadFolder, fileName);
+
+    return new Promise((resolve, reject) => {
+        // Ensure the download folder exists
+        if (!fs.existsSync(downloadFolder)) {
+            fs.mkdirSync(downloadFolder);
+        }
+
+        const file = fs.createWriteStream(filePath);
+        https.get(url, (response) => {
+            if (response.statusCode !== 200) {
+                reject(`Failed to get '${url}' (${response.statusCode})`);
+                return;
+            }
+            response.pipe(file);
+            file.on('finish', () => {
+                file.close();
+                resolve(filePath);
+            });
+        }).on('error', (err) => {
+            fs.unlink(filePath, () => reject(err.message));
+        });
+    });
+}
+
+
+run()
```

Comment:
GH is complaining about the missing extra line (even if it's not that important for me 😅)

Reply:
Done